cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1549,13 +1549,19 @@ public Map<ProgramRunId, RunRecordDetail> getRuns(ApplicationId applicationId, f
   @Nullable
   public RunRecordDetail getRun(ProgramRunId programRun) throws IOException {
     // Query active run record first
-    RunRecordDetail running = getUnfinishedRuns(Collections.singleton(programRun)).get(programRun);
-    // If program is running, this will be non-null
-    if (running != null) {
-      return running;
+    Map<ProgramRunId, RunRecordDetail> unfinishedRunsMap = getUnfinishedRuns(Collections.singleton(programRun));
```

Comment:
Was getUnfinishedRuns() changed in some other PR? It's very error prone (as evidenced by this bug) to take a collection of ProgramRunId but return other ProgramRunIds. Can we change it to take ProgramReference and RunId instead?

Reply:
There are two `getUnfinishedRuns`, the other one is in dogwatch loghandler. Have to admit it's very error prone since we're not passing the version ID and fetching runs ignoring it. `ProgramRunReference` is form the other PR. I can add a TODO to use it once the other PR is merged, since it's major blocking us now

Reply:
It's more work refactoring the whole thing related to versionless runs

Reply:
will update this part in https://github.com/cdapio/cdap/pull/14671